test_streamy.py

```diff
@@ -33,3 +36,8 @@ def test_error(self):
         self.stream_fails('}')
         self.stream_fails('2,,')
         self.stream_fails('[2,]')
+
+
+class NotSeekableTest(StreamyTest):
```

Comment:
I just deleted a bunch of comments :-D because I came up with a better way - I think:

    def _stream(self, s):
        s = io.StringIO(fp)
        s.seekable = lambda: False
        return streamy.stream(s)

and then get rid of the `_force_not_seekable` idea entirely.

There are two reasons for this:

1. It tests a tiny bit more of the target code by having a case where `s.seekable()` returns `False`.
2. It means that the library code doesn't need to know about the testing code.